lib/patch.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+var fs = require('fs');
+var url = require('url');
+
+module.exports = function(req, res) {
+  var urlpieces = req.url.split('/');
+  var id = urlpieces[urlpieces.length-1];
+  var filename = id + ".json";
+  var fullPath = './data/' + filename;
+  var file;
+  var input;
+
+  req.on('data', function(data) {
+    input = data.toString('utf-8');
+    input = JSON.parse(input);
+  });
+  req.on('end', function() {
+    fs.open(fullPath, 'r+', function(err) {
+      if(err) {
+        console.log('file doesn\'t exist');
+        res.writeHead(404);
+        res.end();
+      } else {
+        fs.readFile(fullPath, function(err, data) {
```

Comment:
this is starting to look a little "pyramid of doom"y, You should probably try to refactor it.